### DISCUSSION TITLE
Starlette Slowness Issue

### DISCUSSION THREAD

**Discussion by arper9** [encode/starlette#2156] [Q&A]
Hi Team, I have developed the backend service using FastAPI and the Starlette middleware. I am experiencing a slowness issue with the Starlette send activity when I have captured the profiling information using Sentry. Please find the reference for the same.

Screenshot:
![image](https://github.com/encode/starlette/assets/82801641/796404fb-3543-4bdb-afb8-09285bec7b7f)

Kindly help us with the solution to come out of this issue.

Thanks in advance. :-)

**Comment by Kludex** [comment]
How does the middleware looks like?

**Reply by arper9** [reply]
The middleware is taking longer than usual time or the expected time. we are using the default configuration of the middleware.

**Reply by Kludex** [reply]
I mean how is the implementation?

**Reply by arper9** [reply]
Please find the implementation code.
`from starlette_prometheus import metrics, PrometheusMiddleware
import sentry_sdk

#sentry
sentry_sdk.init(
    dsn="https://**************.ingest.sentry.io/4505249061863424",
    environment=os.environ.get('ENV'),
    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    # We recommend adjusting this value in production,
    traces_sample_rate=1.0,
    _experiments={
      "profiles_sample_rate": 1.0,  # for versions before 1.17.0
    },
    release="*****@V0.0.2",
    # If you wish to associate users to errors (assuming you are using
    # django.contrib.auth) you may enable sending PII data.
    send_default_pii=True,
)

app.add_middleware(PrometheusMiddleware)
app.add_route("/metrics", metrics)`

We are using the default configuration of the sentry sdk. This particular service is hosted in Google Cloud Run Service.

**Reply by Kludex** [reply]
Can you provide an [MRE](https://stackoverflow.com/help/minimal-reproducible-example)?
